fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/BooleanLiteral.java

```diff
@@ -32,11 +32,27 @@ public class BooleanLiteral extends Literal {
 
     private final boolean value;
 
-    public BooleanLiteral(boolean value) {
+    private BooleanLiteral(boolean value) {
         super(BooleanType.INSTANCE);
         this.value = value;
     }
 
+    public static BooleanLiteral of(boolean value) {
+        if (value) {
+            return TRUE;
+        } else {
+            return FALSE;
+        }
+    }
+
+    public static BooleanLiteral of(Boolean value) {
```

Comment:
i think we do not need this function, since java could do auto unboxing

Reply:
Literal also has `of` function, if this don't exist, `Literal of` will call itself which is dead loop.